src/test/java/com/hillel/items_exchange/service/UserServiceTest.java

```diff
@@ -58,13 +69,59 @@ void testUpdateUserEmail_WhenDataCorrect_Successfully() {
         String message = userService.updateUserEmail(userChangeEmailDto, userWithOldPassword);
 
         assertEquals(getMessageSource("changed.user.email"), message);
+        assertEquals(NEW_USER_EMAIL, userWithOldPassword.getEmail());
+        verify(userRepository).saveAndFlush(userWithOldPassword);
+    }
+
+    @Test
+    void testSelfDeleteRequest_WhenDataCorrect_Successfully() {
+        userService.selfDeleteRequest(userWithOldPassword);
+
+        assertEquals(DELETED, userWithOldPassword.getStatus());
+        verify(userRepository).saveAndFlush(userWithOldPassword);
+    }
+
+    @Test
+    void testPermanentlyDeleteUsers_ShouldDeleteRequiredUsers() {
+        User shouldBeDeleted = createUserForDeleting(DELETED,
+                LocalDateTime.now().minusDays(numberOfDaysToKeepDeletedUsers + 1));
+        User shouldNotBeDeleted0 = createUserForDeleting(ACTIVE, LocalDateTime.now());
+        User shouldNotBeDeleted1 = createUserForDeleting(DELETED,
+                LocalDateTime.now().minusDays(numberOfDaysToKeepDeletedUsers - 1));
+        User shouldNotBeDeleted2 = createUserForDeleting(ACTIVE,
+                LocalDateTime.now().minusDays(numberOfDaysToKeepDeletedUsers + 1));
+        List<User> users = List.of(shouldBeDeleted, shouldNotBeDeleted0, shouldNotBeDeleted1, shouldNotBeDeleted2);
```

Comment:
Why don't you extract lines 86 - 93 into a new method because it's just creation of a pre-contitions?

Reply:
![image](https://user-images.githubusercontent.com/49348448/110349160-19d8da80-803b-11eb-8572-88689745f058.png)
I think my version is more readable, but I'll change it if you like.